profiles/schema.py

```diff
@@ -770,6 +773,11 @@ def mutate_and_get_payload(cls, root, info, **input):
         return CreateProfileMutation(profile=profile)
 
 
+@field_validations(
```

Comment:
I was thinking about if the decorator style is a bit too indirect. I mean we have the class right here. Maybe we could write the functions used by the graphene validator by ourselves. Something like:

```python
class VerifiedPersonalInformationAddressInput(graphene.InputObjectType):
    street_address = graphene.String(
        description="Street address with possible house number etc. Max length 1024 characters."
    )
    postal_code = graphene.String(
        description="Postal code. Max length 1024 characters."
    )
    post_office = graphene.String(
        description="Post office. Max length 1024 characters."
    )

    @staticmethod
    def validate_street_address(value):
        return model_field_validation(EncryptedAddress, "street_address", value)

    @staticmethod
    def validate_postal_code(value):
        return model_field_validation(EncryptedAddress, "postal_code", value)

    @staticmethod
    def validate_post_office(value):
        return model_field_validation(EncryptedAddress, "post_office", value)
```

Then the only special thing would be the `model_field_validation` function. Which could be something simple like:

```python
def model_field_validation(model_class, field_name, value):
    field = model_class._meta.get_field(field_name)

    try:
        return field.clean(value, model_class())
    except DjangoValidationError as e:
        raise _multi_validation_error_from_django_validation_error(e)
```

What do you think?

Reply:
That's a possibility too. Having the decorator available doesn't mandate that it needs to be used or that all fields in an input object type should be included in it. It's still ok to write one's own `validate_field_name` function and it can do something special. I wrote the decorator as a helper shorthand for the case that I thought would be quite a common one.

Looking at your alternative code does reveal though, that writing the validator functions by hand isn't that much more code compared to using the decorator so it is a valid question does the decorator really produce value. Perhaps not that much.

I'm starting to lean towards your simpler solution. Let's try it out.

Reply:
Updated the last commit, now without the decorator.